### ISSUE TITLE
Regression with ZipArchive after libzip4 upgrade

### ISSUE THREAD

**Issue by aboks** [oerdnj/deb.sury.org#1521]
**Frequently asked questions**

- [X] I have read [Frequenty Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)

**Describe the bug**
After libzip4 was upgraded to 1.7.3 (as mentioned in #1518), we see regressions where PHP's ZipArchive extension gives an error result rather than creating a new zip file when using `ZipArchive::open()` with the `ZipArchive::CREATE` flag.

**To Reproduce**
```php
<?php
$zipfile_path = tempnam(sys_get_temp_dir(), "zip");
$zip_archive = new \ZipArchive();
var_dump($zip_archive->open($zipfile_path, \ZipArchive::CREATE));
```
When running this script with `libzip4` version `1.7.3-1+0~20210114.10+debian10~1.gbp4c125d` this script gives output `int(19)`, which is `ZipArchive::ER_NOZIP` ("Not a zip archive"). 

**Expected behavior**
Expected output is `bool(true)`.

**Distribution (please complete the following information):**
 - OS: Debian Buster
 - Architecture: amd64
 - Repository: packages.sury.org

**Package(s) (please complete the following information):**
```
$ apt-cache policy libzip4
libzip4:
  Installed: 1.7.3-1+0~20210114.10+debian10~1.gbp4c125d
  Candidate: 1.7.3-1+0~20210114.10+debian10~1.gbp4c125d
  Version table:
 *** 1.7.3-1+0~20210114.10+debian10~1.gbp4c125d 500
        500 http://HTTPS///packages.sury.org/php buster/main amd64 Packages
        100 /var/lib/dpkg/status
     1.5.1-4 500
        500 http://ftp.nl.debian.org/debian buster/main amd64 Packages

$ apt-cache policy php7.4-zip
php7.4-zip:
  Installed: 7.4.14-1+0~20210112.34+debian10~1.gbpaa175a
  Candidate: 7.4.14-1+0~20210112.34+debian10~1.gbpaa175a
  Version table:
 *** 7.4.14-1+0~20210112.34+debian10~1.gbpaa175a 500
        500 http://HTTPS///packages.sury.org/php buster/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Additional context**
- When downgrading `libzip4` back to version `1.5.1-4` shipped with Debian Buster and running the script again, it gives the expected output. 
- When adding the `\ZipArchive::OVERWRITE` flag (i.e. replacing the second parameter to `open` by `\ZipArchive::CREATE | \ZipArchive::OVERWRITE`), it also works with the newer `libzip4` version.

**Comment by todeveni** [comment]
Changed in libzip 1.6.0:
> Do not accept empty files as valid zip archives any longer.

**Comment by oerdnj** [comment]
This is not a packaging issue, but upstream issue, closing the issue.

**Comment by aboks** [comment]
For those running into the mentioned issue and finding this thread: PHP was supposed to have a backward compatibility workaround for libzip >= 1.6.0, but that didn't work correctly: https://bugs.php.net/bug.php?id=80648. A fix for that problem has now been merged and should be released with the upcoming PHP 7.4.15/8.0.2. Also note that the BC workaround gives deprecation warnings as of PHP 8; the future-proof way is to provide the `\ZipArchive::TRUNCATE` flag yourself (see https://github.com/php/php-src/pull/6625#issuecomment-763636899).
